packages/scope-manager/src/scope/ScopeBase.ts

```diff
@@ -232,7 +232,7 @@ abstract class ScopeBase<
     block: TBlock,
     isMethodDefinition: boolean,
   ) {
-    const upperScopeAsScopeBase = upperScope as Scope;
```

Comment:
I'm surprised TS allowed this - the reason this cast was there was because `upperScope` is typed as `TUpper` - and there was a lot of weird behaviour that stemmed from it being a generic instead of the concrete type.

I think there was also some weirdness due to the cyclic nature of the files? I don't fully remmeber though

Reply:
🤷